astropy/wcs/tests/test_pickle.py

```diff
@@ -103,16 +98,35 @@ def test_wcs():
 
 class Sub(wcs.WCS):
     def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
         self.foo = 42
 
 
 def test_subclass():
-    wcs = Sub()
-    s = pickle.dumps(wcs)
-    with pytest.warns(FITSFixedWarning):
-        wcs2 = pickle.loads(s)
+    wcs1 = Sub()
+    wcs1.foo = 45
```

Comment:
Tsk tsk, but 42 is the answer to everything!
```suggestion
    wcs1.foo = 42
```